### PR TITLE
Fix misc issues reported by Deepscan

### PR DESCRIPTION
**Pull request by znarf**

### CODE REVIEW

File: components/edit-collective/RequestVirtualCardModal.js [hunk 1]
```diff
@@ -100,7 +100,7 @@ const RequestVirtualCardModal = props => {
       if (!values.purpose) {
         errors.purpose = 'Required';
       }
-      if (!values.notes && values.notes?.lenght > 10) {
```
Comment (znarf) [inline components/edit-collective/RequestVirtualCardModal.js:103]: Typo

File: components/edit-collective/Form.js [hunk 2]
```diff
@@ -789,7 +789,7 @@ class EditCollectiveForm extends React.Component {
       ],
     };
 
-    Object.keys(this.fields).map(fieldname => {
```
Comment (znarf) [inline components/edit-collective/Form.js:792]: map is not the best syntax to iterate on an array when nothing is returned

File: components/I18nAddressFields.js [hunk 3]
```diff
@@ -299,7 +299,7 @@ const I18nAddressFields = ({
           label={fieldLabel}
           info={fieldInfo}
           value={value?.[fieldName]}
-          required={required === false ? false : !Object.keys(data?.optionalLabels).includes(fieldName)}
+          required={required === false ? false : !Object.keys(data?.optionalLabels || {}).includes(fieldName)}
```
Comment (znarf) [inline components/I18nAddressFields.js:302]: If you use includes, make sure it's an object.

Should it be an array actually?

File: components/expenses/ProcessExpenseButtons.js [hunk 4]
```diff
@@ -310,7 +310,7 @@ const ProcessExpenseButtons = ({
           onDelete={onDelete}
         />
       )}
-      {displaySecurityChecks && expense?.securityChecks?.length && (
+      {displaySecurityChecks && expense?.securityChecks?.length > 0 && (
```
Comment (znarf) [inline components/expenses/ProcessExpenseButtons.js:313]: Not the proper way to check length in such react condition

File: components/host-dashboard/CreatePendingOrderModal.tsx [hunk 5]
```diff
@@ -326,17 +326,6 @@ const CreatePendingContributionForm = ({ host, onClose, error, edit }: CreatePen
...
-  const receiptTemplateTitles = [];
-  if (receiptTemplates?.default?.title?.length > 0) {
-    receiptTemplateTitles.push({
-      value: 'default',
-      label: receiptTemplates?.default?.title,
-    });
-  }
-  if (receiptTemplates?.alternative?.title?.length > 0) {
-    receiptTemplateTitles.push({ value: 'alternative', label: receiptTemplates?.alternative?.title });
-  }
```
Comment (znarf) [inline components/host-dashboard/CreatePendingOrderModal.tsx:339]: Dead code not used.

File: pages/confirmEmail.js [hunk 6]
```diff
@@ -33,13 +33,13 @@ class ConfirmEmailPage extends React.Component {
...
     if (!this.props.loadingLoggedInUser) {
-      return this.triggerEmailValidation();
+      this.triggerEmailValidation();
     }
   }
 
   componentDidUpdate() {
     if (!this.state.validationTriggered && !this.props.loadingLoggedInUser) {
-      return this.triggerEmailValidation();
+      this.triggerEmailValidation();
```
Comment (znarf) [inline pages/confirmEmail.js:42]: No need to return anything here

File: pages/search.js [hunk 7]
```diff
@@ -169,7 +169,7 @@ class SearchPage extends React.Component {
 
   constructor(props) {
     super(props);
-    this.onClick = this.onClick.bind(this);
```
Comment (znarf) [inline pages/search.js:172]: Unnecessary bind, this.onClick is already bound.

File: pages/order.tsx [hunk 8]
```diff
@@ -634,7 +634,7 @@ export default function OrderPage(props) {
                   })
                 )}
               </Box>
-              {hasProcessButtons(order?.permissions) && (
+              {hasProcessButtons(order.permissions) && (
```
Comment (znarf) [inline pages/order.tsx:637]: `order` is not null checked earlier and `permissions` needs to be set later.

**Review by Betree — APPROVED** [review]

**Pull request merged**